libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -978,85 +976,106 @@ ArraySchema ArrowAdapter::tiledb_schema_from_arrow_schema(
         bool isattr = true;
 
         for (int64_t i = 0; i < index_column_schema->n_children; ++i) {
-            auto achild = index_column_array->children[i];
-            auto schild = index_column_schema->children[i];
-            auto col_name = schild->name;
-            if (strcmp(child->name, col_name) == 0) {
-                if (ArrowAdapter::arrow_is_var_length_type(child->format)) {
-                    type = TILEDB_STRING_ASCII;
-                }
+            if (strcmp(child->name, index_column_schema->children[i]->name) ==
+                0) {
+                if (strcmp(child->name, "soma_geometry") == 0 &&
```

Comment:
Maybe we should `const std::string` some of these in `utils/common.h` like "soma_geometry", "tiledb__internal__", "__min", and "__max"